packages/nodes-base/nodes/QuestDb/QuestDb.node.ts

```diff
@@ -182,22 +228,34 @@ export class QuestDb implements INodeType {
 			const returnFields = this.getNodeParameter('returnFields', 0) as string;
 
 			const queries : string[] = [];
-			items.map(item => {
-				const columns = Object.keys(item.json);
+			items.map((item, index) => {
+				const columns = this.getNodeParameter('columns', index) as string;
+				
+				let columnNames: string[];
+				if (columns !== '') {
+					columnNames = columns.split(',').map(columnName => columnName.trim());
+				} else {
+					columnNames = Object.keys(item.json);
+				}
 
-				const values : string = columns.map((col : string) => {
-					if (typeof item.json[col] === 'string') {
-						return `\'${item.json[col]}\'`;
-					} else {
+				const values : string = columnNames.map((col : string) => {
+					if (typeof item.json[col] === 'number') {
+						// Skip quotes only for numbers. formats like dates should be quoted.
 						return item.json[col];
+					} else {
+						return `\'${item.json[col]}\'`;
 					}
 				}).join(',');
 
-				const query = `INSERT INTO ${tableName} (${columns.join(',')}) VALUES (${values});`;
+				const query = `INSERT INTO ${tableName} (${columnNames.join(',')}) VALUES (${values});`;
 				queries.push(query);
 			});
 
-			await db.any(pgp.helpers.concat(queries));
+			for(let i = 0; i < queries.length; i++) {
+				// We have to insert lines one by one and wait
+				// Otherwise we might get `table is busy` errors.
+				await db.any(queries[i]);
+			}
```

Comment:
this might be more readable, but did not test it!
```suggestion
			for(const query of queries) {
				// We have to insert lines one by one and wait
				// Otherwise we might get `table is busy` errors.
				await db.any(query);
			}
```